pkg/agent/containerd/nvidia.go

```diff
@@ -0,0 +1,61 @@
+// +build linux
+
+package containerd
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/rancher/k3s/pkg/agent/templates"
+)
+
+// findNvidiaContainerRuntimes returns a list of nvidia container runtimes that
+// are available on the system. It checks install locations used by the nvidia
+// gpu operator and by system package managers. The gpu operator installation
+// takes precedence over the system package manager installation.
+// The given fs.FS should represent the filesystem root directory. If nil,
+// os.DirFS("/") is used.
+func findNvidiaContainerRuntimes(fsys fs.FS) []templates.ContainerdRuntimeConfig {
+	rootFS := fsys
+	if rootFS == nil {
+		rootFS = os.DirFS("/")
+	}
+
+	// Check these locations in order. The GPU operator's installation should
+	// take precedence over the package manager's installation.
+	locationsToCheck := []string{
+		"usr/local/nvidia/toolkit", // Path when installing via GPU Operator
+		"usr/bin",                  // Path when installing via package manager
+	}
+
+	// Fill in the binary location with just the name of the binary,
+	// and check against each of the possible locations. If a match is found,
+	// set the location to the full path.
+	potentialRuntimes := []templates.ContainerdRuntimeConfig{
+		{
+			Name:        "nvidia",
+			RuntimeType: "io.containerd.runc.v2",
+			BinaryName:  "nvidia-container-runtime",
+		},
+		{
+			Name:        "nvidia-experimental",
+			RuntimeType: "io.containerd.runc.v2",
+			BinaryName:  "nvidia-container-runtime-experimental",
+		},
+	}
+	foundRuntimes := []templates.ContainerdRuntimeConfig{}
+RuntimeLoop:
```

Comment:
I think the convention is to make the label all caps? But if you make this a map instead of a list, you might not need to do this anyway, as the map itself will handle duplicates.
```suggestion
RUNTIME:
```